test_requests.py

```diff
@@ -1224,6 +1224,7 @@ def test_equality(self):
         del othercid['spam']
         assert cid != othercid
         assert cid == {'spam': 'blueval', 'eggs': 'redval'}
+        assert cid.__eq__(object()) == NotImplemented
```

Comment:
If you have to access private methods to test something, you shouldn't be testing it.

I thought that returning `NotImplemented` triggered a `NotImplementedError` but that doesn't happen in all cases it seems.

```py
cid == object()
# => False
```

But I don't think we strictly need this assertion anyway. You're testing an implementation detail, not a public part of the API (which is all we should ever test).